plugins/Hosting/js/components/resizeDialog.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import { Map } from 'immutable'
+import path from 'path'
+
+const ResizeDialogModal = ({ shouldShowResizeDialog, resizePath, resizeSize, initialSize, actions }) => {
+	const handleSettingInput = (e) => actions.updateModal('resizeSize', e.target.value)
+	const hideResizeDialog = (newSize) => actions.hideResizeDialog(Map({ path: resizePath, size: newSize }))
+	const closeResizeDialog = () => hideResizeDialog(0)
+	const handleSubmit = () => {
+		if (resizeSize >= 35 && resizeSize !== initialSize.toString()) {
+			hideResizeDialog(resizeSize)
+		}
+	}
+
+	const handleSettingKeyDown = (e) => {
+		if (e.keyCode === 13) {
+			handleSubmit()
+			e.preventDefault()
+		}
+	}
+
+	return (
+		<div className={'hosting-options-modal modal' + (shouldShowResizeDialog ? '': ' hidden')}>
+			<form className="hosting-options modal-message" onSubmit="">
+				<div className="close-button" onClick={closeResizeDialog}>
+					X
+				</div>
+
+				<h3>Resize &quot;{path.basename(resizePath)}&quot;</h3>
+				<p>
+					<label>Size in GB (Min is 35 GB)</label>
+					<input type="number" onChange={handleSettingInput} onKeyDown={handleSettingKeyDown} value={resizeSize} min="35"></input>
+				</p>
+				<span className={'error' + ( resizeSize < 35 ? '' : ' hidden' )}>Storage folder must be at least 35 GB.</span>
```

Comment:
I think ternaries for styling is fine.  Looks like this always displays the modal component, but sets `hidden` in css if it shouldn't be shown.  Not sure how this differs from ternarying the component rendering (`{shouldRender ? <Component /> : null`})